src/binding.hpp

```diff
@@ -1,17 +1,59 @@
 #include <v8-cppgc.h>
+#include <v8-isolate.h>
 #include <v8-message.h>
 
 /**
  * Types defined here will be compiled with bindgen
  * and made available in `crate::binding` in rust.
  */
 
-namespace {
+extern "C" {
+class RUST_ExternalOneByteString
+    : public v8::String::ExternalOneByteStringResource {
+ public:
+  using RustDestroyFn = void (*)(char*, size_t);
+  // bindgen doesn't support generating bindings for inline
+  // constructors/functions
+  RUST_ExternalOneByteString(char* data, int length, RustDestroyFn rustDestroy,
+                             v8::Isolate* isolate);
 
-class RustObj;
+  ~RUST_ExternalOneByteString() {
+    (*_rustDestroy)(_data, _length);
+    _isolate->AdjustAmountOfExternalAllocatedMemory(
+        -static_cast<int64_t>(-_length));
+  }
 
+  const char* data() const { return _data; }
+
+  size_t length() const { return static_cast<size_t>(_length); }
+
+ private:
+  char* const _data;
+  const size_t _length;
+  RustDestroyFn _rustDestroy;
+  v8::Isolate* _isolate;
+};
+
+class RustObj final : public cppgc::GarbageCollected<RustObj> {
```

Comment:
Couldn't figure out how to get this to work without just copying the RustObj definition over. The definition here was obviously causing conflicts when I included the header in `binding.cc` (so that I could implement the constructor there)